Lib/test/test_asyncio/test_selector_events.py

```diff
@@ -766,14 +766,48 @@ def test_write_sendmsg_partial(self):
         data = memoryview(b'data')
         self.sock.sendmsg = mock.Mock()
         # Sent partial data
-        self.sock.sendmsg.return_value = len(data) // 2
+        self.sock.sendmsg.return_value = 2
 
         transport = self.socket_transport(sendmsg=True)
         transport._buffer.append(data)
         self.loop._add_writer(7, transport._write_ready)
         transport._write_ready()
         self.assertTrue(self.sock.sendmsg.called)
         self.assertTrue(self.loop.writers)
+        self.assertEqual(list_to_buffer([b'ta']), transport._buffer)
+
+    @unittest.skipUnless(selector_events._HAS_SENDMSG, 'no sendmsg')
+    def test_write_sendmsg_half_buffer(self):
+        data = [memoryview(b'data1'), memoryview(b'data2')]
+        self.sock.sendmsg = mock.Mock()
+        # Sent partial data
+        self.sock.sendmsg.return_value = 2
+
+        transport = self.socket_transport(sendmsg=True)
+        transport._buffer.extend(data)
+        self.loop._add_writer(7, transport._write_ready)
+        transport._write_ready()
+        self.assertTrue(self.sock.sendmsg.called)
+        self.assertTrue(self.loop.writers)
+        self.assertEqual(list_to_buffer([b'ta1', b'data2']), transport._buffer)
```

Comment:
This feels like a Yoda condition.

Reply:
> Yoda condition.

That's a new word for me, TIL. Are you suggesting to change the order of arguments or something else?